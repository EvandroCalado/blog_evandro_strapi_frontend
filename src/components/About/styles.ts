import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    ${Title} {
      font-size: ${theme.font.sizes['4xl']};
    }
`}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};

    &::first-letter {
      font-size: ${theme.font.sizes.xl};
      font-weight: bold;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`

    & > img {
      width: 35rem;
      height: 35rem;
      overflow: hidden;
      border-radius: ${theme.spacings.md};
      object-fit: cover;
    }
`}
`;
