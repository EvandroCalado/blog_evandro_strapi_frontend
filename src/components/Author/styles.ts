import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    margin: ${theme.spacings.xl} auto;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
      width: 30rem;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 22rem;
    height: 17rem;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
    margin-right: ${theme.spacings.md};

    & img {
      object-fit: cover;
    }

    @media screen and ${theme.media.lteOrEqSmall} {
      width: 15rem;
      margin-bottom: ${theme.spacings.lg};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    @media screen and ${theme.media.lteOrEqSmall} {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.sm} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
`}
`;
