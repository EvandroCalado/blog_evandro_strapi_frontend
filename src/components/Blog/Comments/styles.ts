import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;

    iframe {
      color: ${theme.colors.text};
    }
`}
`;
