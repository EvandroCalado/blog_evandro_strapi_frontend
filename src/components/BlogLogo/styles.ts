import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Title = styled.h1`
  text-transform: uppercase;
  letter-spacing: 0.7rem;
  font-weight: 400;
`;
