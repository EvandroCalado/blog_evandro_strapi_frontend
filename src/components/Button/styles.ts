import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondary};
    color: ${theme.colors.neltral1};
    outline: none;
    border: none;
    padding: ${theme.spacings.md} ${theme.spacings.xl};
    font-size: ${theme.font.sizes.sm};
    font-weight: bold;
    transition: ${theme.transitions.faster};
    cursor: pointer;

    &:disabled {
      background-color: ${theme.colors.neltral5};
      pointer-events: none;
    }

    &:hover {
      background-color: ${theme.colors.secondaryHover};
    }

    &:active {
      background-color: ${theme.colors.secondaryActive};
    }
`}
`;
