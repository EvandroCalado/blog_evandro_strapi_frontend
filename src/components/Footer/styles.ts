import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    text-transform: uppercase;
    background-color: ${theme.colors.secondary};
    padding: ${theme.spacings.md} ${theme.spacings.lg};
    color: ${theme.colors.neltral1};
`}
`;

export const Socials = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['6xl']};
    margin: 0 auto;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: ${theme.spacings.md};
    border-bottom: 1px solid ${theme.colors.neltral1};
    margin-bottom: ${theme.spacings.md};

      & a {
        transition: ${theme.transitions.faster};

        &:hover {
          color: ${theme.colors.primary};
      }

      & svg {
        width: 4rem;
      }
  `}
`;

export const Developer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['6xl']};
    margin: 0 auto;
    display: flex;
    align-items: center;
    justify-content: space-between;

      @media ${theme.media.lteOrEqSmall} {
        flex-direction: column;
        text-align: center;
      }
  `}
`;
