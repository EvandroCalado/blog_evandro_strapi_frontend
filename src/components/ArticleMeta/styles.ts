import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['7xl']};
    margin: auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    & > span > a {
      font-weight: bold;
      transition: ${theme.transitions.faster};
      margin-left: ${theme.spacings.sm};

      &:hover {
        text-decoration: underline;
      }
    }

    ${Title} {
      width: 100%;
      max-width: ${theme.frameSizes['4xl']};
      text-align: center;
    }
`}
`;

export const Image = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 40rem;
    object-fit: cover;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
  `}
`;

export const ArticleContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
    }
`}
`;

export const DateContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: end;
    justify-content: center;
    gap: ${theme.spacings.lg};
`}
`;
