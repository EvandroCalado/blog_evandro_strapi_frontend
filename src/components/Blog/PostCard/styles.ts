import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: ${theme.spacings.lg} ${theme.spacings.md};

    & > a {
      width: 100%;
      height: 20rem;
      margin-bottom: ${theme.spacings.md};
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 100%;
    margin-bottom: ${theme.spacings.md};

    & img {
      object-fit: cover;
      border-radius: ${theme.spacings.sm};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.md};

    & > a {
      transition: ${theme.transitions.faster};

      &:hover {
        text-decoration: underline;
        filter: invert(.2);
      }
    }

    ${Title} {
      overflow: hidden;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.xs} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 4;
`}
`;

export const DateContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const Date = styled.span`
`;
