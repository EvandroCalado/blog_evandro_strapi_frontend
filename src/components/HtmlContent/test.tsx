import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import HtmlComponent from './';

describe('<HtmlComponent />', () => {
  it('should render with default values', () => {
    renderTheme(<HtmlComponent html="<b>test</b>" />);
    expect(screen.getByText('test')).toBeInTheDocument();
    expect(screen.getByText('test')).toHaveStyle('font-weight: bold');
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(<HtmlComponent html="<b>test</b>" />);
    expect(container).toMatchSnapshot();
  });
});
