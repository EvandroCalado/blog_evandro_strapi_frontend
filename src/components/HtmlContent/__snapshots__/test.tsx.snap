// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlComponent /> should render a snapshot 1`] = `
.c0 code[class*="language-"],
.c0 pre[class*="language-"] {
  color: #f8f8f2;
  background: none;
  text-shadow: 0 1px rgba(0, 0, 0, 0.3);
  font-family: Consolas,Monaco,'Andale Mono','Ubuntu Mono',monospace;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;
  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.c0 pre[class*="language-"] {
  padding: 1em;
  margin: .5em 0;
  overflow: auto;
  border-radius: 0.3em;
}

.c0 :not(pre)>code[class*="language-"],
.c0 pre[class*="language-"] {
  background: #282a36;
}

.c0 :not(pre)>code[class*="language-"] {
  padding: .1em;
  border-radius: .3em;
  white-space: normal;
}

.c0 .token.comment,
.c0 .token.prolog,
.c0 .token.doctype,
.c0 .token.cdata {
  color: #6272a4;
}

.c0 .token.punctuation {
  color: #f8f8f2;
}

.c0 .namespace {
  opacity: .7;
}

.c0 .token.property,
.c0 .token.tag,
.c0 .token.constant,
.c0 .token.symbol,
.c0 .token.deleted {
  color: #ff79c6;
}

.c0 .token.boolean,
.c0 .token.number {
  color: #bd93f9;
}

.c0 .token.selector,
.c0 .token.attr-name,
.c0 .token.string,
.c0 .token.char,
.c0 .token.builtin,
.c0 .token.inserted {
  color: #50fa7b;
}

.c0 .token.operator,
.c0 .token.entity,
.c0 .token.url,
.c0 .language-css .token.string,
.c0 .style .token.string,
.c0 .token.variable {
  color: #f8f8f2;
}

.c0 .token.atrule,
.c0 .token.attr-value,
.c0 .token.function,
.c0 .token.class-name {
  color: #f1fa8c;
}

.c0 .token.keyword {
  color: #8be9fd;
}

.c0 .token.regex,
.c0 .token.important {
  color: #ffb86c;
}

.c0 .token.important,
.c0 .token.bold {
  font-weight: bold;
}

.c0 .token.italic {
  font-style: italic;
}

.c0 .token.entity {
  cursor: help;
}

.c1 {
  width: 100%;
  max-width: 80rem;
  margin: 0 auto;
  padding: 2rem 0;
}

.c1 a {
  font-weight: bold;
  transition: all 200ms ease-in-out;
}

.c1 a:hover {
  text-decoration: underline;
}

.c1 a:visited {
  color: #3b82f6;
}

.c1 h2,
.c1 h3,
.c1 h4,
.c1 h5,
.c1 h6,
.c1 p {
  margin: 2rem 0;
}

.c1 img {
  max-width: 100%;
}

.c1 pre {
  border-radius: 0.5rem;
}

.c1 .image {
  background: #020415;
  line-height: 0;
  margin: 2rem 0;
}

.c1 .image figcaption {
  font-size: 1.4rem;
  color: #ffffff;
  padding: 0.5rem;
  text-align: center;
  line-height: 1.3;
}

.c1 .image-style-side {
  max-width: 50%;
  float: none;
  margin: 2rem;
}

.c1 ul,
.c1 ol {
  margin: 2rem 3rem;
}

.c1 blockquote {
  border-left: 0.5rem solid;
  font-style: italic;
  padding-left: 2rem;
  margin: 2rem 0;
}

@media screen and (min-width: 576px) {
  .c1 .image-style-side {
    float: right;
    max-width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <b>
        test
      </b>
    </div>
  </div>
</div>
`;
